spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -1483,7 +1483,7 @@ Here is an example where an application uses `@Query` to provide a partial query
 ----
 @Repository
 public interface CustomerRepository extends BasicRepository<Customer, Long> {
-  @Query("SELECT o FROM Customer o WHERE (o.totalSpent / o.totalPurchases > ?1)")
+  @Query("WHERE (o.totalSpent / o.totalPurchases > ?1)")
```

Comment:
```suggestion
  @Query("WHERE (totalSpent / totalPurchases > ?1)")
```

Reply:
I mean this is just really hostile to the user. Why would you want to make the user do this every time?

Reply:
> I mean this is just really hostile to the user. Why would you want to make the user do this every time?

Did this comment end up in the wrong place?  It is attached to my correction of query language where the indentification variable was incompletely removed.

Reply:
Oh I misunderstood. I thought it was also about parentheses.

JDQL does not have identification variables, and they're also now optional in JPQL.

Reply:
Fixed.

Reply:
> JDQL does not have identification variables, and they're also now optional in JPQL.

The problem is that the change halfway removed them from a particular query.  The query used to look like this:

```
  @Query("SELECT o FROM Customer o WHERE (o.totalSpent / o.totalPurchases > ?1)")
```

You removed the part that defines the variable `o`, but not its usage:
```
@Query("WHERE (o.totalSpent / o.totalPurchases > ?1)")
```

So my proposed change is that we need to also remove `o.` from the rest of the query.